spec/models/user_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+    describe ".find_or_create_user" do
+      let(:user_attributes) { HashWithIndifferentAccess.new(user.attributes) }
```

Comment:
I had to go read up on it, `HashWithIndifferentAccess`. I never knew about it. Thanks for sharing.